themes-default/slim/views/restart.mako

```diff
@@ -24,23 +52,22 @@ window.app = new Vue({
 </%block>
 <%block name="content">
 <h2>{{ $route.meta.header }}</h2>
-<div default-page="${sbDefaultPage}" current-pid="${sbPID}" class="messages">
+<div>
     <div id="shut_down_message">
         Waiting for Medusa to shut down:
-        <img src="images/loading16-${app.THEME_NAME}.gif" height="16" width="16" id="shut_down_loading" />
-        <img src="images/yes16.png" height="16" width="16" id="shut_down_success" style="display: none;" />
+        <loading-spinner v-if="status === 'shutting_down'"></loading-spinner>
+        <yes-image v-else></yes-image>
     </div>
-    <div id="restart_message" style="display: none;">
+    <div id="restart_message" v-if="status === 'initializing' || status === 'restarted'">
         Waiting for Medusa to start again:
-        <img src="images/loading16-${app.THEME_NAME}.gif" height="16" width="16" id="restart_loading" />
-        <img src="images/yes16.png" height="16" width="16" id="restart_success" style="display: none;" />
-        <img src="images/no16.png" height="16" width="16" id="restart_failure" style="display: none;" />
+        <loading-spinner v-if="status !== 'restarted'"></loading-spinner>
+        <yes-image v-if="status === 'restarted'"></yes-image>
+        <no-image v-if="status === 'failed'"></no-image>
     </div>
-    <div id="refresh_message" style="display: none;">
-        Loading the default page:
-        <img src="images/loading16-${app.THEME_NAME}.gif" height="16" width="16" id="refresh_loading" />
+    <div id="refresh_message" v-if="status === 'restarted'">
+        Loading the default page: <loading-spinner></loading-spinner>
     </div>
-    <div id="restart_fail_message" style="display: none;">
+    <div id="restart_fail_message" v-if="status === 'failed'">
```

Comment:
This is never called on the old setup so should we just remove it?

The old js is below.
```js
let currentPid = $('.messages').attr('current-pid');
const defaultPage = $('.messages').attr('default-page');
const checkIsAlive = setInterval(() => {
    // @TODO: Move to API
    $.get('home/is_alive/', data => {
        if (data.msg.toLowerCase() === 'nope') {
            // If it's still initializing then just wait and try again
            $('#restart_message').show();
        } else if (currentPid === '' || data.msg === currentPid) {
            $('#shut_down_loading').hide();
            $('#shut_down_success').show();
            currentPid = data.msg;
        } else {
            clearInterval(checkIsAlive);
            $('#restart_loading').hide();
            $('#restart_success').show();
            $('#refresh_message').show();
            setTimeout(() => {
                window.location = defaultPage + '/';
            }, 5000);
        }
    }, 'jsonp');
}, 100);
```

Reply:
Using the websocket is an option too, however you'd still need to know if the app is fully initialized. Currently this is the check with the process ID - if the app is not ready no PID is sent back.

Reply:
We could wait for an "ok" message on the socket?

Reply:
If the client browser isn't connected it might not receive the message,
I don't think it has the same reliability.